build-system/tasks/build.js

```diff
@@ -100,6 +100,7 @@ build.flags = {
   coverage: '  Adds code coverage instrumentation to JS files using istanbul.',
   version_override: '  Overrides the version written to AMP_CONFIG',
   watch: '  Watches for changes in files, re-builds when detected',
+  esm: '  Do not transpile down to ES5',
```

Comment:
I think we need to consume `argv.esm` in the call to `esbuild`, right? IIRC, @jridgewell tried doing this in [`e1c028c` (#32699)](https://github.com/ampproject/amphtml/pull/32699/commits/e1c028cd341f4301e4018e262abd630b6c3cff12#r577259088) and had to revert in [`85ab0b8` (#32699)](https://github.com/ampproject/amphtml/pull/32699/commits/85ab0b81d48209534df8fa8c23a151e84f6ad231).

Reply:
ah! that is different. that is literally about outputting modules, as opposed to "esm build", which simply reduces transpilation.

Reply:
For reference, the unminified dist build goes from `799K` to `770K` with `--esm` flag

Reply:
I've reduced the transpilation further to align better with what we do in `--compiled` mode. Now it is down to `754K`, and should be even faster to run :)

Reply:
Aha, I now see. This gets `babel` to do something different because of `argv.esm`, while `esbuild` does its regular thing.